numpy/lib/financial.py

```diff
@@ -651,7 +651,7 @@ def irr(values):
     """
     res = np.roots(values[::-1])
     mask = (res.imag == 0) & (res.real > 0)
-    if res.size == 0:
+    if res.size == 0 or not mask.any():
```

Comment:
Yea really shouldn't do this before coffee in the morning!

Sorry for making so many trivial mistakes